pandas/io/packers.py

```diff
@@ -591,19 +591,21 @@ def decode(obj):
                           name=obj[u'name'])
 
     elif typ == u'series':
+        name = obj[u'name']
+        index = obj[u'index']
+        klass = globals()[obj[u'klass']]
+
         dtype = dtype_for(obj[u'dtype'])
         pd_dtype = pandas_dtype(dtype)
-        np_dtype = pandas_dtype(dtype).base
-
-        index = obj[u'index']
-        result = globals()[obj[u'klass']](unconvert(obj[u'data'], dtype,
-                                                    obj[u'compress']),
-                                          index=index,
-                                          dtype=np_dtype,
-                                          name=obj[u'name'])
         tz = getattr(pd_dtype, 'tz', None)
+
+        arr = unconvert(obj[u'data'], dtype, obj[u'compress'])
+
         if tz:
+            result = klass(arr, index=index, dtype=pd_dtype.base, name=name)
```

Comment:
I don't think you need to do this duplication, create the result, the if a tz, convert, no? (IOW you can just pass ``pd_dtype``, ``pd_dtype.base`` is not very useful